apps/autoscheduler/AutoSchedule.cpp

```diff
@@ -113,28 +120,35 @@ vector<vector<int64_t>> generate_tilings(const vector<int64_t> &s, int d, int fa
                     t.back() = outer;
                     result.push_back(t);
                 }
-                for (int outer = 1; outer <= s[d]; outer *= factor) {
-                    int inner = (s[d] + outer - 1) / outer;
-                    if (is_one && outer == 1) continue;
-                    if (is_full && outer == s[d]) continue;
-                    // Stop when we get into the regime covered by the loop above.
-                    if (outer > 1 && inner < max_inner * 2) break;
-                    // Or when the wasted compute gets too bad.
-                    if (inner * outer * 7 > s[d] * 8) break;
-                    t.back() = outer;
-                    result.push_back(t);
-                }
 
-                // The sequence above (in terms of the inner loop) goes 1 2 4 8 16 ...
-                // but 3 is an important inner tiling factor for matrix multiply ops.
-                int inner3 = (d == vector_dim) ? 3*vector_size : 3;
-                int outer3 = (s[d] + inner3 - 1) / inner3;
-                if (factor == 2 && inner3 < s[d] && outer3 < s[d] && outer3 > 1) {
-                    if (inner3 * outer3 * 7 <= s[d] * 8) {
-                        t.back() = outer3;
+                // Additional tiling options may have inner extents that are not
```

Comment:
None of this should be necessary anymore. compute_here has already created an inner loop equal to the vector size, so any tiling produced here is already implicitly a multiple of the vector size

Reply:
Ah, I said this because I think that the vector loop should be gpu_threads, but I see below that you've just picked the first three loops you find inside the gpu_blocks loops. This is going to create far too many threads a lot of the time due to data dependencies. The loop structure should be blocks -> serial loops -> threads